### DISCUSSION TITLE
How to click control `CheckBox` from code?

### DISCUSSION THREAD

**Discussion by judexgauss** [AvaloniaUI/Avalonia#16862] [Q&A]
I have a `CheckBox control`
When I'm trying to raise an event of click (need this for ui tests purposes) there are no click and no errors.

Code snippet i'm using:  
```
control.RaiseEvent(new RoutedEventArgs(CheckBox.ClickEvent));
```

**Comment by stevemonaco** [comment]
I'm not sure why this doesn't work. As an alternative, Avalonia has a [Headless Platform](https://docs.avaloniaui.net/docs/concepts/headless/) which is intended for application UI testing that doesn't require native OS integrations (like window positioning tests which Avalonia itself should be responsible for).

It has extension methods for keyboard and mouse input, including `MouseDown` and `MouseUp` which you can wrap into something more convenient to use.

https://github.com/AvaloniaUI/Avalonia/blob/master/src/Headless/Avalonia.Headless/HeadlessWindowExtensions.cs
